### PR TITLE
Add "about us" and "contact us" endpoints

### LINKED ISSUE #22: Implement "about us" and "contact" page
Proposal: Modelate about us content

### LINKED ISSUE #66: Identify if a job offer permits remote working
# Backgrounds

At this stage, there is not a checkbox to indicate if a job vacancies is a remote job.

# Issue

Nowadays, many people need to know if a job is remote. People can apply to a job in another city or country if it's a remote job or just work remotely because it offers a more flexible lifestyle.

95682136-f58be580-0be3-11eb-8dd3-39da63273941

# Potential approaches

## Solution 1

Do nothing. This info can be included on Job Description field.

_be free to vote with :-1:_

## Solution 2 

Add "Is remote" on `Jobs` as a new field:

_be free to vote with :+1:_

```python
class Job(models.Model):
    # ...
    is_remoting = models.BooleanField("Is remoting", default=False, )
```

## Other alternative solutions.

_please, be free to post your own as a new comment._

# More screenshots

image

### LINKED ISSUE #68: Job Salary: should be a range?
# Background

The job vacancies has a "Salary" field:

```python
class Job(models.Model):
    # ...
    salary = models.IntegerField(default=0, blank=True)
```

Screenshot:

Job screenshot with salary field

# Issue

Usually, companies post vacancies without salary or with a salary range:

Vacancies screenshot with salary range

And applicants search offers for a minimum annual salary:

Sample searching for a minimum annual salary

# Potential approaches

## Solution 1

Do nothing. This info can be included on Job Description field.

_be free to vote with :-1:_

## Solution 2 

Change salary by a salary range (annual_salary_max and annual_salary_min or external package like [inclusive-django-range-fields](https://github.com/Hipo/inclusive-django-range-fields)). Also, maybe allow null on values.

_be free to vote with :+1:_

## Other alternative solutions.

_please, be free to post your own as a new comment._

### PR DESCRIPTION
**Pull request by fullonic**
## Description

Please describe your pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue. When linking to an issue, please use `refs #...` in the description of the pull request.

Fix #22 
Fix #66 
Fix #68

Please provide enough information so that others can review your pull request:
This adds two new api endpoints: 
- /api/about-us
- /api/contact-us

Also adds a new html page Contact us
- /contact-us

### CODE REVIEW

File: jobsapp/api/views/common.py
Comment (fullonic) [inline jobsapp/api/views/common.py:50]: @francescarpi which information should we share here?
@XaviTorello may you help me here providing the correct information for each field?

My approach here will be create a new model, for example `AboutUs`, and add it to the admin page, so we can change the information from there. It's okay or there is a better approach?

File: jobsapp/api/views/common.py
Comment (fullonic) [inline jobsapp/api/views/common.py:60]: @francescarpi which information should we share here?
@XaviTorello may you help me here providing the correct information for each field?

My approach here will be create a new model, for example `AboutUs`, and add it to the admin page, so we can change the information from there. It's okay or there is a better approach?

File: jobs/settings.py
Comment (francescarpi) [inline jobs/settings.py:300]: You can use `env.bool("EMAIL_USE_TLS", default=False)`

File: jobsapp/api/views/common.py
Comment (francescarpi) [inline jobsapp/api/views/common.py:46]: The dictionary's key (message) should not be translated.

File: jobsapp/api/views/common.py
Comment (francescarpi) [inline jobsapp/api/views/common.py:54]: We need to search if exists any dajngo's package for manage page's contents. 
We should create about's page content in admin.

File: jobsapp/forms.py
Comment (francescarpi) [inline jobsapp/forms.py:45]: The form and serializer (ContactSerializer) should have same definition (max_lengths, etc)

**Review by francescarpi — COMMENTED** [review]
Hi! Good job. 
I've put some comments.

File: jobsapp/api/views/common.py
Comment (fullonic) [reply jobsapp/api/views/common.py:54]: Something like django flat pages? https://docs.djangoproject.com/en/3.1/ref/contrib/flatpages/

Comment (francescarpi) [reply jobsapp/api/views/common.py:52]: Yes, could be a good idea. Come on.

**Comment by francescarpi** [comment]
I was thinking would be better to split this PR because it contains quite different logical. 

Maybe in this one you can only keep the contact api.

Latter, in a other PR, put the django contact view.

And finally, in other one, put the about page with django flatpages.

It will be better. Are you agree?

Good job!

File: jobs/settings.py
Comment (jbagot) [inline jobs/settings.py:304]: This one is a secret. Should be without default. This way you are forcing to add a password in the environment vars when we will do the deploy.

File: jobsapp/api/views/common.py
Comment (jbagot) [inline jobsapp/api/views/common.py:39]: This is not wrong, but since we are using django rest framework you can use the APIViews, GenericViews or Viewsets. Creating a class with a method `post` or `create` depending on what class you will use will be nicer and more consistent with the rest of the views that we have.

File: jobs/settings.py
Comment (jbagot) [reply jobs/settings.py:304]: When you will do that, run `make test` and won't work, you will need to add this secret in the docker-compose.testing.yml like the postgres password and so on.

**Review by jbagot — CHANGES_REQUESTED** [review]

File: jobs/settings.py
Comment (fullonic) [reply jobs/settings.py:304]: Thanks, I was not aware of it. :+1:

File: jobsapp/api/views/common.py
Comment (fullonic) [reply jobsapp/api/views/common.py]: Yep, makes sense.

**Comment by fullonic** [comment]
> I was thinking would be better to split this PR because it contains quite different logical.
> 
> Maybe in this one you can only keep the contact api.
> 
> Latter, in a other PR, put the django contact view.
> 
> And finally, in other one, put the about page with django flatpages.
> 
> The reasons it's also because in this PR #129 are a lot changes related with templates (for example has been moved in each app)
> 
> It will be better. Are you agree?
> 
> Good job!
Hi, okay, sounds good to me. So, would be better close this PR and open a new one for each issue you pointed out? Should I wait for #129 be merge and then do a PR to Contact View Form?

**Comment by francescarpi** [comment]
> > I was thinking would be better to split this PR because it contains quite different logical.
> > Maybe in this one you can only keep the contact api.
> > Latter, in a other PR, put the django contact view.
> > And finally, in other one, put the about page with django flatpages.
> > The reasons it's also because in this PR #129 are a lot changes related with templates (for example has been moved in each app)
> > It will be better. Are you agree?
> > Good job!
> 
> Hi @francescarpi , okay, sounds good to me. So, would be better close this PR and open a new one for each issue you pointed out? Should I wait for #129 be merge and then do a PR to Contact View Form?

Hi! Yes, I prefer close this one, and create one for each...

You can start with contact's api.

My PR (#129) probably be merged this afternoon.

Thanks and sorry for this changes...

**Comment by fullonic** [comment]
It's okay for me ;)
So I'm closing this one and split it into 3. I will open now a PR for `api/contact-us` and once #129 is merged, I will open the two missing ones.

**Pull request closed without merging**
